apepay/__init__.py

```diff
@@ -74,11 +82,67 @@ def validators(self) -> List[Validator]:
                         )
                     )
                 )
-            except ContractLogicError:
+
+            except (ContractLogicError, DecodingError):
+                # NOTE: Vyper returns no data if not a valid index
                 break
 
         return validators
 
+    def _convert_to_address(self, item: _ValidatorItem) -> str:
+        if isinstance(item, Validator):
+            return item.contract.address
+        elif isinstance(item, ContractInstance):
+            return item.address
+        else:
+            return item
+
+    def set_validators(
+        self,
+        validators: List[_ValidatorItem],
+        **txn_kwargs,
+    ) -> ReceiptAPI:
+        if len(validators) > 20:
+            raise
```

Comment:
an empty raise is not valid Python.
There is no exception on the stack.

Try this in a python inpretter:

```python
raise
```

**The only time you can use an empty raise is when there is an exception on the stack - to raise that exception and preserver the exception stack**
like this:

```python
try:
  asfasdf
except Exception:
   print("asdfaasdfasdf")
   raise  # Raise the exception we caught in the except.
```

An empty `raise` by itself in the middle of nowhere is not valid tho, you will get this error:

```
  File "<stdin>", line 1, in <module>
RuntimeError: No active exception to reraise
```

Reply:
Maybe you thought you were writing Vyper?

Reply:
I resolved this.